graph.py

```diff
@@ -10,11 +12,54 @@
     log.error(f"Package {PACKAGE} not found. Please 'pip install {PACKAGE}'")
 
 
+class Filter:
+    """Class responsible for removing items with disqualifying values."""
+
+    def __init__(self, filter_type, filter_function):
+        self._filter_type = filter_type
+        self._filter_fun = filter_function
+
+    def run(self, value, items):
+        """Filter out items. Filter chosen is picked at runtime."""
+        if isinstance(value, self._filter_type):
+            return filter(self._filter_fun(value), items)
+
+        raise TypeError(f"Expected type: {self._filter_type}")
+
+
 class KytosGraph:
     """Class responsible for the graph generation."""
 
     def __init__(self):
         self.graph = nx.Graph()
+        self._filter_fun_dict = {}
+
+        def filter_leq(metric):  # Lower values are better
+            return lambda x: (lambda y: y[2].get(metric, x) <= x)
+
+        def filter_geq(metric):  # Higher values are better
+            return lambda x: (lambda y: y[2].get(metric, x) >= x)
+
+        def filter_eeq(metric):  # Equivalence
+            return lambda x: (lambda y: y[2].get(metric, x) == x)
+
+        self._filter_fun_dict["ownership"] = Filter(
+            str, filter_eeq("ownership"))
+        self._filter_fun_dict["bandwidth"] = Filter(
+            (int, float), filter_geq("bandwidth"))
+        self._filter_fun_dict["priority"] = Filter(
+            (int, float), filter_geq("priority"))
+        self._filter_fun_dict["reliability"] = Filter(
+            (int, float), filter_geq("reliability"))
+        self._filter_fun_dict["utilization"] = Filter(
+            (int, float), filter_leq("utilization"))
+        self._filter_fun_dict["delay"] = Filter(
+            (int, float), filter_leq("delay"))
+        self._path_fun = nx.all_shortest_paths
+
+    def set_path_fun(self, path_fun):
+        """Set the shortest path function."""
+        self._path_fun = path_fun
```

Comment:
This method is only called by tests. Is it really necessary?

Reply:
It is not required and will be removed.